### PR TITLE
Challenge create using UI

### PR DESCRIPTION
**Pull request by RishabhJain2018**
UI for Challenge creation using UI.

### CODE REVIEW

File: frontend/src/css/modules/challenge-create.scss [hunk 1]
```diff
@@ -4,3 +4,7 @@
 	margin-bottom: 20px;
 	margin-left: 11px;
 }
+
+.add-margin {
+	margin-bottom: 30px;
+}
```
Comment (deshraj) [inline frontend/src/css/modules/challenge-create.scss]: Add a newline here.

**Comment by deshraj** [comment]
Tried to test on my local machine and it doesn't seem to work. Here are some comments:

- Datetime picker not working
- Evaluation script will accept only .zip file
- Blocked after the first step of challenge creation
- If some fields are left empty, then show the error that fields are required. 

Please fix these.

**Comment by deshraj** [comment]
screen shot 2017-10-03 at 6 45 31 pm

Please see the above screenshot for the error on the console.

**Comment by deshraj** [comment]
@isht3: would you mind reviewing this pull request? Please comment wherever you feel things could have been written in a better way or so. Looking forward to seeing your first review. :)

File: frontend/src/css/modules/challenge-create.scss [hunk 2]
```diff
@@ -4,3 +4,7 @@
 	margin-bottom: 20px;
 	margin-left: 11px;
 }
+
+.add-margin {
```
Comment (aka-jain) [inline frontend/src/css/modules/challenge-create.scss]: @RishabhJain2018 please follow proper naming for classes, add-margin is way too common.

File: frontend/src/css/modules/create-challenge-using-ui.scss [hunk 3]
```diff
@@ -0,0 +1,145 @@
+.checkbox-field {
```
Comment (aka-jain) [inline frontend/src/css/modules/create-challenge-using-ui.scss]: Change name or use nested classes please.

File: frontend/src/css/modules/create-challenge-using-ui.scss [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+	margin-top: 20px;
+
+}
+
+.tooltip {
+	word-wrap: keep-all;
+	max-width: 100px;
+}
+
+.reviewScreen {
```
Comment (aka-jain) [inline frontend/src/css/modules/create-challenge-using-ui.scss]: Don't use camel case here (if not needed)

File: frontend/src/js/directives/directives.js [hunk 5]
```diff
@@ -344,3 +344,155 @@
...
+            link: function() {
+                //do nothing      
+            },
+            templateUrl: 'dist/views/web/create-challenge-using-ui/edit-create-challenge-using-ui/edit-create-challenge.html',
+            transclude: true,
+            restrict: 'EA'
+        };
+        return directive;
+    }
+})();
```
Comment (aka-jain) [inline frontend/src/js/directives/directives.js]: Why do you use directives here, simple partials can work here. @RishabhJain2018

File: frontend/src/js/route-config/route-config.js [hunk 6]
```diff
@@ -172,6 +172,18 @@
...
 
+        var create_challenge_using_ui = {
+            name: "web.create-challenge-using-ui",
+            parent: "web",
+            url: "/create-challenge-using-ui",
+            templateUrl: baseUrl + "/web/create-challenge-using-ui/create-challenge-using-ui.html",
+            title: 'Create Challenge',
+            controller: 'CreateChallengeUsingUiCtrl',
+            controllerAs: 'createChallengeUsingUi',
+            // redirectTo: "web.challenge-create.challenge-list",
```
Comment (aka-jain) [inline frontend/src/js/route-config/route-config.js]: Remove commented code.

File: frontend/src/views/web/challenge-create.html [hunk 7]
```diff
@@ -25,6 +25,29 @@
...
+                    <div>OR</div>
+                </h5>
+            </div>
+        </div>
+    </div>
+    <div class="row">
+        <div class="col s12 m9 offset-m3">
+            <div class="add-margin">
+                <h5 class="mons-med-font">
+                    <div>Fill challenge creation form</div>
```
Comment (aka-jain) [inline frontend/src/views/web/challenge-create.html]: Remove empty divs. @RishabhJain2018

File: frontend/src/views/web/challenge-create.html [hunk 8]
```diff
@@ -25,6 +25,29 @@
...
+                </h5>
+            </div>
+            <div class="add-margin">
+                <h6 class="mons-med-font">
+                    <a href="http://google.co.in" target="_blank"> <div>View detailed procedure</div></a>
```
Comment (aka-jain) [inline frontend/src/views/web/challenge-create.html]: Empty divs.

File: frontend/src/views/web/create-challenge-using-ui/create-challenge-using-ui.html [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+
+<review-screen>
+</review-screen>
+
+<submission-screen>
+</submission-screen>
+
+
+<edit-challenge>
+</edit-challenge>
```
Comment (aka-jain) [inline frontend/src/views/web/create-challenge-using-ui/create-challenge-using-ui.html]: Use partials here.

File: frontend/src/js/controllers/createChallengeUsingUi.js [hunk 10]
```diff
@@ -0,0 +1,613 @@
+// Invoking IIFE for create challenge using ui
```
Comment (aka-jain) [inline frontend/src/js/controllers/createChallengeUsingUi.js]: Change the name of file to createChallengeUsingUiCtrl and be ensure to add 'Ctrl' at the end of each file of controller.

File: frontend/src/js/controllers/createChallengeUsingUi.js [hunk 11]
```diff
@@ -0,0 +1,613 @@
...
+        vm.step1e = false; // variable to store edit form
+        vm.step2 = false;
+        vm.step2e = false;
+        vm.step3 = false;
+        vm.step3e = false;
+        vm.step4 = false;
+        vm.step4e = false;
+        vm.step5 = false;
+        vm.step5e = false;
+        vm.step6 = false;
```
Comment (aka-jain) [inline frontend/src/js/controllers/createChallengeUsingUi.js]: You can create an array or object for step variables. If possible do it so,. @RishabhJain2018

**Comment by aka-jain** [comment]
@deshraj @RishabhJain2018 when it will be ready to merge?

**Comment by deshraj** [comment]
@RishabhJain2018 can you please resolve the conflicts? I can test it then

**Comment by RishabhJain2018** [comment]
@deshraj Yes sure !

**Comment by deshraj** [comment]
@RishabhJain2018 any update on this?

**Comment by isht3** [comment]
@deshraj I'm sorry for being this late but I somehow must've missed this notification in my feed and my E-mails. I will check this today.

**Comment by deshraj** [comment]
@RishabhJain2018 can I review this now?

File: frontend/src/views/web/challenge-create.html [hunk 12]
```diff
@@ -23,8 +23,31 @@
...
```
Comment (spyshiv) [inline frontend/src/views/web/challenge-create.html]: @RishabhJain2018 change hyperlink url here

**Comment by deshraj** [comment]
@RishabhJain2018 were you able to resolve conflicts and incorporate all the comments/suggestions?

**Comment by deshraj** [comment]
@RishabhJain2018 Any updates on this? You should prioritize this and get this done ASAP. It has been more than 10 months since you have been working on this feature.

**Comment by RishabhJain2018** [comment]
@deshraj Since this PR is too old. I have incorporated all it's comments in the new PR. Hence, I am closing this one.

**Pull request closed without merging**